### ISSUE TITLE
New UpDate MKS TINYBEE Marlin 2.1.x Fix Bug

### ISSUE THREAD

**Issue by flishy2532** [makerbase-mks/MKS-TinyBee#39]
This New Marlin for MKS TINYBEE Am Try TEST And Fix Now Work With TMC2209 v1.0 v2.0 with Out Prolem

 SOON COME v 2.1.x bugfix and v 2.0.9.2

This FirmWare TEST Full WORK

MKS_TINYBEE  Marlin 2.1.x
-MKS v 3.0 RBG full color support 
-WiFi support SDCard support 
-TMC2209 x4 ( v1.0-v2.0) 
###This Config Fix and TEST By. BIRD Engineer ThaiLand ###
contract us : pig_466@hotmail.com 
Line ID : flishy Thank You 
Hope Have fun And Make Item With 3D Print ^.^  _/\_

**Comment by wirebeard** [comment]
Do you using UART for stepper driver?

**Comment by flishy2532** [comment]
am use tmc and interpola config

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by 3DSmitty** [comment]
@flishy2532 In Configuration.h you have your drivers set to A4988 not TMC2209 so you are not using UART.

**Comment by flishy2532** [comment]
am change to 2209standalone

**Comment by 3DSmitty** [comment]
Ok TMC2209_STANDALONE is not UART. This means you have to set your driver current physically using the pot on the driver instead of being able to do it on the LCD or via M906.

**Comment by 3DSmitty** [comment]
Also interpolation is enabled by default via the hardware.

**Comment by flishy2532** [comment]
how to fix layershift on delta printer ? i have this problem on mks tinybee

**Comment by 3DSmitty** [comment]
most likely either your vref current on the tmc drivers are not set correctly or acceleration in marlin is too high.

**Comment by flishy2532** [comment]
am set vref 850 am try drop accel

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by flishy2532** [comment]
![20220624_012530.jpg](https://user-images.githubusercontent.com/61683774/175368867-f3f8359c-6ba1-43b3-94b4-9863d2d1560c.jpg)


LayerShift aw

**Comment by fred4537** [comment]
Hello 
can you try babystepping without probe with fimware 2.0.9.2
because without probe the babystepping no work with me
thank

**Comment by flishy2532** [comment]
ok let me test

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by flishy2532** [comment]
> Hello 
> can you try babystepping without probe with fimware 2.0.9.2
> because without probe the babystepping no work with me
> thank

v 2.0.9.2 have bug babystepping use other version or download bugfix

**Comment by fred4537** [comment]
ok thank you

**Comment by flishy2532** [comment]
no problem did you use delta printer?

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by fred4537** [comment]
no I use Anet A8 .
Marlin bugfix 2.1 is good for you

**Comment by flishy2532** [comment]
2.1.x have bug surface move too 🥺

**Comment by 3DSmitty** [comment]
@flishy2532 Are you able to control your part cooling fan with M106? I am not able to turn on the part cooling fan in Marlin 2.1

**Comment by flishy2532** [comment]
> @flishy2532 Are you able to control your part cooling fan with M106? I am not able to turn on the part cooling fan in Marlin 2.1

am use auto fan on cura 5.1.0 turn on fan cooling at layer 2

**Comment by flishy2532** [comment]
> @flishy2532 Are you able to control your part cooling fan with M106? I am not able to turn on the part cooling fan in Marlin 2.1

![20220801_214626.jpg](https://user-images.githubusercontent.com/61683774/182177920-ea278a73-5577-4f4b-8a58-5abff33bbd44.jpg)

![20220801_214643.jpg](https://user-images.githubusercontent.com/61683774/182177939-b0ed0f77-d980-42ab-b80a-06e842333b94.jpg)

![20220801_215157.jpg](https://user-images.githubusercontent.com/61683774/182177955-3ed5337b-be38-4ebf-833d-f5188b28cd7d.jpg)

![20220801_215210.jpg](https://user-images.githubusercontent.com/61683774/182177967-2032509a-9be8-4b20-9645-8dddcf28b07a.jpg)

**Comment by 3DSmitty** [comment]
@flishy2532  Got it thank you! Had to enable
`#define FAN_SOFT_PWM`

Did you have to change the `#define SOFT_PWM_SCALE 0` settings at all?

**Comment by flishy2532** [comment]
it work now?

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by 3DSmitty** [comment]
Yes thank you! Did you have to change the #define SOFT_PWM_SCALE 0 setting at all?

**Comment by flishy2532** [comment]
> Yes thank you! Did you have to change the #define SOFT_PWM_SCALE 0 setting at all?

only  #define SOFT_PWM_SCALE 0

**Comment by flishy2532** [comment]
am print this on my kossel linear plus
![20220801_225748.jpg](https://user-images.githubusercontent.com/61683774/182190530-d67dfa98-f81c-4280-9446-d2313a497001.jpg)

![20220801_225804.jpg](https://user-images.githubusercontent.com/61683774/182190566-1638fe4a-c7de-4a86-b28e-89114d1fd03d.jpg)

![20220801_225824.jpg](https://user-images.githubusercontent.com/61683774/182190591-0472360d-7773-43d0-af81-737f261aaf16.jpg)

**Comment by 3DSmitty** [comment]
Nice! I put my Tinybee in a Monoprice MP10.
![IMG_20220801_114141032](https://user-images.githubusercontent.com/51137582/182191445-a7d74203-5a7b-4c00-bbeb-381691460e37.jpg)
!

**Comment by flishy2532** [comment]
wow all work now?

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by flishy2532** [comment]
this fame look like my ender3 pro

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by 3DSmitty** [comment]
> wow all work now?

Yes so far starting to tune in Marlin.

> this fame look like my ender3 pro

Yes similar frame. Bed is AC Mains powered 300x300.

**Comment by flishy2532** [comment]
this can print via wifi on cura 5 mks wifi plug in?

รับ Outlook สำหรับ Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by 3DSmitty** [comment]
No idea I use Prusa Slicer

**Comment by fred4537** [comment]
hello 3DSmitty
  your similar ender is ok with Marlin 2.1
 have  you got  MKS lcd 12864  mini V3 ?
your babystepping is ok and your print layer 0.1 is ok

**Comment by 3DSmitty** [comment]
@fred4537 Yes I am running Marlin 2.1 on the Tinybee on my Monoprice MP10. Yes it is the MKS MINI12864 V3 LCD. I haven't tried babystepping yet and my layer height for printing is 0.2mm. I have printed a few XYZ cubes so far that came out pretty good, I still need to fine tune the acceleration etc as I am getting some ghosting.

**Comment by fred4537** [comment]
@3DSmitty you can compile marlin 2.1 with visual studio code now ?
because I don't compile marlin 2.1 .
I can just compile  marlin 2.1.1 , but i have not got the lcd. MKS lcd is good with marlin 9.2 and now is black.

**Comment by 3DSmitty** [comment]
@fred4537 the MKS MINI12864 V3 LCD is working fine for me. Attached is my Marlin config files maybe that will help
[Configuration.txt](https://github.com/makerbase-mks/MKS-TinyBee/files/9312528/Configuration.txt)
[Configuration_adv.txt](https://github.com/makerbase-mks/MKS-TinyBee/files/9312529/Configuration_adv.txt)
.

**Comment by fred4537** [comment]
ok thanks

**Comment by fred4537** [comment]
@3DSmitty 
Hello
when I compile marlin 2.1 , it's good but the LCD is black. I see your config.
you have maybe  a file marlin 2.1 for tinybee.
 can you share your full marlin file. 
thanks

**Comment by flishy2532** [comment]
> @3DSmitty 
> Hello
> when I compile marlin 2.1 , it's good but the LCD is black. I see your config.
> you have maybe  a file marlin 2.1 for tinybee.
>  can you share your full marlin file. 
> thanks

yes i have both 2.1 and 2.1.x bugfix

**Comment by flishy2532** [comment]
this 2.1.x bugfix with lcd mks v3.0![20220813_144057.jpg](https://user-images.githubusercontent.com/61683774/184474305-d6996b18-b64b-43d4-8ed5-b3b201bafe44.jpg)

**Comment by fred4537** [comment]
yes I believe you. I see your machine 
but I don’t understand why lcd lights up with  MKS tinybee Marlin (version 2.0.9)  , and if I configure marlin 2.1 or marlin  2.1.x bugfix of the web site "https://github.com/MarlinFirmware/Marlin and I define MKS_MINI_12864_V3 the LCD.
the Lcd does not light up

**Comment by 3DSmitty** [comment]
@fred4537 Those are the only two files from Marlin that I configured. All other files are stock from Marlin. Are you sure you 

`#define NEOPIXEL_LED`

It has to be enabled for MKS MINI12864 V3 LCD

**Comment by 3DSmitty** [comment]
@flishy2532 Did you forget spring cleaning this year? Your desk would make me insane, haha.

**Comment by fred4537** [comment]
thank you 
it was #define NEOPIXEL_LED
MKS tinybee marlin (2.0.9) of the siteweb makerbse is disable and it was ok. STRANGE
When I power now printer 3d where I can config the color /default

**Comment by 3DSmitty** [comment]
@fred4537 oh good glad you figured it out. Happy Printing!

**Comment by flishy2532** [comment]
https://drive.google.com/file/d/19MKl1yQN89sEd2PhazUwl3XpRqco5tuS/view?usp=sharing

this exe file fix layer shift bug  change g-code flover in cura to Rearap is bester dont see layer shift

https://drive.google.com/file/d/16XyWPW_-BK43I3r1XOchSx3ZkwAzy-y7/view?usp=sharing

this exe file fix layer shift bug  change g-code flover in cura to Rearap is bester dont see layer shift

**Comment by flishy2532** [comment]
![20221013_220802.jpg](https://user-images.githubusercontent.com/61683774/195699730-baad5f5b-d572-4871-9c99-1ef72b4922f8.jpg)

![20221013_220633.jpg](https://user-images.githubusercontent.com/61683774/195699785-36f0c16a-7ed1-4fd0-90c4-e68aab155bfc.jpg)

![20221013_220639.jpg](https://user-images.githubusercontent.com/61683774/195699874-4250e578-5cf0-440f-a018-a100e23b4636.jpg)

![20221013_220725.jpg](https://user-images.githubusercontent.com/61683774/195699964-b3d8e13a-c26a-437f-a12c-e4b2d82efe0c.jpg)

**Comment by AngelSerra** [comment]
I know this post is old, but from the photos you need a hot and clean bed with clean and smooth glass, not what you have there, the problem is that the filament does not stick to the hot bed
